.github/workflows/test-integration-unreleased.yml

```diff
@@ -48,10 +48,22 @@ jobs:
           path: opensearch/distribution/archives/linux-tar/build/distributions
           key: ${{ steps.get-key.outputs.key }}
 
+      - name: Set up JDK 17
```

Comment:
So we need 2 JDKs to be installed in case of 11: 11 + 17, and only one in all other cases. Adding  `java-runtime`  would lead to duplicate `actions/setup-java@v4` runs, isn't it?

Reply:
Like this:

```
      matrix:
        entry:
          - { opensearch_ref: '1.x', java: 11 }
          - { opensearch_ref: '2.x', java: 11, java-runtime: 17 }
          - { opensearch_ref: '2.x', java: 17 }
          - { opensearch_ref: '2.x', java: 21 }
          - { opensearch_ref: 'main', java: 11, java-runtime: 17 }
          - { opensearch_ref: 'main', java: 17 }
          - { opensearch_ref: 'main', java: 21 }
```

Then below you do `if: matrix.java-runtime` and `java-version: ${{ matrix.java-runtime }}`.

Reply:
Looks better @dblock ?